src/test/java/org/jenkinsci/plugins/workflow/log/LogStorageTestBase.java

```diff
@@ -202,22 +273,32 @@ protected Map<String, Level> agentLoggers() {
     }
     private static final class RemotePrint extends MasterToSlaveCallable<Void, Exception> {
         private final String message;
-        private final TaskListener listener;
-        RemotePrint(String message, TaskListener listener) {
+        private TaskListener listener;
+        private final RemotingTestVariant variant;
+        RemotePrint(String message, TaskListener listener, RemotingTestVariant variant) {
             this.message = message;
             this.listener = listener;
+            this.variant = variant;
         }
-        @Override public Void call() {
+        @Override public Void call() throws Exception {
             listener.getLogger().println(message);
-            listener.getLogger().flush();
-            return null;
-        }
-    }
-    /** Checking behavior of {@link DelayBufferedOutputStream} garbage collection. */
-    private static final class GC extends MasterToSlaveCallable<Void, Exception> {
-        @Override public Void call() {
-            System.gc();
-            System.runFinalization();
+            if (!variant.disableManualFlush) {
+                listener.getLogger().flush();
+            }
+            switch (variant) {
+                case DelayedAutoFlushOnly -> {
+                    // TODO: It would be better to wait for `DelayBufferedOutputStream.flushBuffer` to run exactly once.
+                    Thread.sleep(DelayBufferedOutputStream.Tuning.DEFAULT.minRecurrencePeriod * 3);
+                }
+                case GcFlushOnly -> {
+                    listener = null;
+                    // Sleeping and calling `System.gc` more than once seem to be necessary for Cleaner to run reliably.
+                    for (int i = 0; i < 3; i++) {
+                        System.gc();
+                        Thread.sleep(1000);
+                    }
+                }
+            }
```

Comment:
Note that in real-world scenarios, where we do not nicely wait after writing each individual line, there is no guarantee of ordering with `GCFlushedOutputStream` or `DelayBufferedOutputStream`, and so you can end up with lines in unexpected orders.